packages/sil-governance/src/routes/dashboard/test/operateDashboard.ts

```diff
@@ -356,3 +367,38 @@ export const checkVotingPanel = (proposal: Element, id: number): void => {
   const participationData = votingPanel.children[2];
   checkParticipationData(participationData, expected);
 };
+
+const clickSortByE2E = async (page: Page): Promise<void> => {
+  await page.click(`input[name="${comparatorLabel}"]`);
+};
+
+const clickTooltipOptionE2E = async (page: Page, option: number): Promise<void> => {
+  await page.click(`div[role="tooltip"] nav > div:nth-child(${option}) p`);
+};
+
+export const sortByExpiryDateE2E = async (page: Page): Promise<void> => {
+  await clickSortByE2E(page);
+  await sleep(1000);
+  await clickTooltipOptionE2E(page, 2);
+};
+
+export const sortByStartDateE2E = async (page: Page): Promise<void> => {
+  await clickSortByE2E(page);
+  await sleep(1000);
+  await clickTooltipOptionE2E(page, 3);
+};
+
+export const sortByVoteE2E = async (page: Page): Promise<void> => {
+  await clickSortByE2E(page);
+  await sleep(1000);
+  await clickTooltipOptionE2E(page, 4);
+};
+
+export const voteYesOnFirstProposalE2E = async (page: Page): Promise<void> => {
+  await page.click(`button[type="submit"]`);
+};
+
+export const voteNoOnFirstProposalE2E = async (page: Page): Promise<void> => {
+  const [txLink] = await page.$x(`//* [@id="proposals"]/div[2]/div[2]/div[1]/form/div[2]/button`);
```

Comment:
Maybe it is easier to simply set id for this button? And click on it based on id?

Reply:
There are 3x9=27 buttons so I don't think the `id` approach is feasible.